index.bs

```diff
@@ -164,16 +167,50 @@ In order to <dfn>create new anchor object</dfn> from |native origin| and |sessio
     1. Return |anchor|.
 </div>
 
+<div class="algorithm" data-algorithm="restore-anchor-from-frame">
+The {{XRAnchor/requestPersistentHandle()}} method, when invoked on an {{XRAnchor}} |anchor|, MUST run the following steps:
+    1. Let |promise| be [=a new Promise=] in the [=relevant realm=] of this {{XRSystem}}.
+    1. If |anchor|'s [=XRAnchor/deleted=] is <code>true</code>, [=/reject=] |promise| with {{InvalidStateError}}, return |promise|, and abort these steps.
+    1. Let |session| be |anchor|'s [=XRAnchor/session=].
+    1. If |session|'s [=XRSession/map of persistent anchors=] contains the value |anchor|, run the following steps:
+        1. Let |uuid| be the key of the mapping of the |anchor| value on |session|'s [=XRSession/map of persistent anchors=].
+        1. [=/Resolve=] |promise| with |uuid|.
+        1. Return |promise|.
+        1. Abort these steps.
+    1. Let |uuid| be the empty string.
+    1. Generate a UUID [[!RFC4122]] as a string and append it to |uuid|.
+    1. Add |uuid| and |anchor| to |session|'s [=XRSession/map of persistent anchors=].
+    1. [=/Resolve=] |promise| with |uuid|.
+    1. Return |promise|.
+</div>
+
+Note: when the {{XRSystem}} creates the unique ID, it MUST be unique and only known to the current origin.
+
+<div class="algorithm" data-algorithm="delete-peristence-from-anchor">
+The {{XRAnchor/deletePersistence()}} method, when invoked on an {{XRAnchor}} |anchor|, MUST run the following steps:
```

Comment:
Do you need to have a persisted XRAnchor instance in your hands in order to be able to "unpersist" it? Should't it be sufficient to pass an UUID to XRSession?

Reply:
I considered putting it on XRSession but this way simplifies the OpenXR implementation. 
I'll investigate how easy this change would be.

Reply:
I agree it makes more sense on session. I updated the spec.